Update with latest changes

tomballgithub opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--  NOTE: In order to check code style locally and avoid having your build rejected by Travis, -->
<!--  run the following commands before you commit: `flake8 .` and `npm run lint`. Fix any -->
<!--  issues they point out. Note also that flake's NOQA is disabled on Travis. -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

sebastienvercammen committed:
Our bans are temporary, not relating to worldly as opposed to spiritual affairs. (#2411)

sebastienvercammen committed:
Skip spawnpoints with missed_count > 5. (#2410)

friscoMad committed:
Avoid converting coordinates to int (#2413)

friscoMad committed:
[DOCS] Update SB and account config documentation (#2356)

* Update SB docs and improve account configuration

Fix my bad english :D

Fix spacing

* Update information abount unbanning and 24h ban

* Address comments and remove the 3 fields limit in CSVs

sebastienvercammen committed:
Add connect timeout for webhooks. (#2418)

Widdin committed:
Fixed broken link in faq.md (#2421)

The speed scheduler was linked to "http://rocketmap.readthedocs.io/en/develop/scanning-method/Speed-Scheduler.html" which isn't a valid site. The correct is with non-capital letters "/speed-scheduler.html"

friscoMad committed:
CLI tool to mass level accounts to level 2 (#2355)

* Add a CLI tool to mass level account to level 2

Rename all folders in lowercase

Add -ams warning and example with command line params

Fix ending too early, fix cancel at any time

Fixed style issues in spawnpoint clustering

Add proxy support requeue failed accounts

Update to latest RM changes, added retry limit, added full locale support and fix docs

* Fix spelling

* Fix spelling for real :D

* I hate Pokémons and Pokéstops

trunksbomb committed:
Added "Exclude Pokemon by Rarity" feature to hide unwanted rarities. (#2387)

* Added "Exclude Pokemon by Rarity" feature to hide unwanted rarities.
Hidden rarities are chosen by a dropdown on the options tab, where each successive option contains all the rarities of the last option.
For example, the "Common" option hides all commons, the "Uncommon" option hides commons and uncommons, etc.

* Formatting fixes to confomr with npm run lint

* Requested changes + cleanup.

* Travis.

Alderon86 committed:
Resurrect Spawn-Scanning (#2238)

* Revive Spawn-Scanning

* removed unneeded lines

* consistency

* Fixing when scanning gyms

* use list.extend instead of list + list

Performace improvement :)

* make it work

* Travis

* use list instead of None

* remove stops & gmys, we dont need them

* Travis...shut up

* added simple speed-limit and clustering for -ss

shamlessly copied from #1476

* consistency update

* Improved spawnpoint handling in SpawnScan.

Removed gym/pokestop from possible locations -  SpawnScan is for scanning spawnpoints only.
Leave models.py "untouched".

* Reworked clustering algorithm to avoid using class Spawnpoint(s).

Moved clustering into schedulers since this should only be used when Spawn Scan is enabled.
In a dense area, with default settings it compressed 102 spawnpoints into 52 which is great.
More refactoring can be done, not sure how to handle old_div, pretty sure we can just do normal division and distance calculations maybe can be handled by equi_rect_dist and soon by cHaversine

* Cleanup and refactoring on SS Clustering.

Removed use of clsmath.py:
 - instead of distance() we're using utils.equi_rect_distance
 - moved intermediate_point() into transform.py
Moved away from old_div - using standard division operator. I think old_div is suppose to make code compatible between python 2 and 3. No need for that here.
Results with different distance function and "different" division function seem very similar. Exactly the same number of cluster as before (112 SP > 55 Clusters - dense area st3)

* Refactoring and some cleanup on SS Clustering algorithm.

* Simplification of search thread logic when waiting for the speed limiter to timeout.

Saves a lot of db updates (I think :P) because the status message kept changing every second.

* Updated/Improved comments.

* Minor text fixes ;)

* Removed '-ssc' parameter and adjusted clustering range to compensate jittering.

Removed "double-check" test because it should never be triggered. In our test we never trigger this error.
'-ssct 240' is how you enable spawnpoint clustering with previous time threshold (240s - 4min).

* fix spawn-scan arguement in config

* fix log message

* changed speedlimiter

* Reworked Speed Limiter into Scheduler.

* More refactoring and optimizations.

Followed most suggestions laid out by other devs and fixed a couple of bugs.
Refactored and renamed parts of the clustering code.

* Improved comments on transform.py:intermediate_point().

* Bug Fix + Speed limiter small improvement

Bug fix: Previously status['last_scanned_date'] wasn't initialized with current date. Now we use (lat, lon) to figure out if it's the first scan or not, and only check the speed limit if location is present.
Improvement: Drop search item if the worker isn't going to reach it in time - saves time wasted on a scan that would be discarded anyway.
Small fix in arg parser - uniformization.

* Updated Docs and removed unnecessary Code and Tools

* Revert change that was out of the scope of this PR.

* (Documentation) Updated commandline.md

* Removed commented code that isn't required.

* Updated docs + Rebase

Documentation now specifies the time unit used in -ssct parameter.
Cleaned up old code that stuck around after rebase.

friscoMad committed:
Remove support for SQLite (#2288)

Move MySQL as part of basic install flow
Update db and code to use bigint instead of strings for encounter ids, spawnpoint ids and cell ids
Add migration code
Remove transaction support
Fix doc problem

Fix rebase

sebastienvercammen committed:
Don't log stack trace for webhook exceptions. (#2426)

* Don't log stack trace for webhook posts.

* Update example config.

Missed in #2418.

sebastienvercammen committed:
Add location marker + mobile Pokémon chunk processing controls in custom.js. (#2427)

* Add location marker controls.

+ Allow mobile chunk settings.

* Add Store setting for isLocationMarkerMovable.

* Clarify updateLocationMarker.

* Add scan location marker settings.

ClarkyKent committed:
Gym info update (#2362)

* [FrontEnd] Added filter for Raids

changes according to tarvis

Bug fix on frontend and have one option instead of two.

make travis happy

added "Show Raids" toggle button to show/hide raids
Take no-raids argument into account to show/hide the toggle button

indentation fix

Error fix, introduced during conflict resolve

lint error fix

Modified according to PR comments:
Now While gyms toggle is active there will be three options
-- Show raids only
-- Show Active raids
-- And pull down menu to filter raid level

If gym toggle is active then it will show all gyms with or without raids
By activating "Show Raids Only" it will hide gyms without Raids
By activating "Show Active Raids" it will hide all raid eggs, leaving only Raids with pokemon.

And there is filter to select the Raid level

commit before rebase
small changes done after PR review

All PR review changes done.

-use of isOngoingRaid function to determine active raids on map.
-check against if gym has raid field null (in case there was never raid scanned/happened on the gym)

using a function to determine raid level which checks if raid field is not null and some lint error fix

* Separated Raid and Gym options.
Now, Raids and gyms switch can be used together or independently and result will vary according to switches enabled.

* - Fixed few bugs where filtering Raid by level was not working.
- Now, GymLabel and gym details will change according to filter(no longer show eggs in Label or in sidebar)
- On gym Marker it will show all pokemons in gym for ongoing raid

* - minor bug fix where Without toggle "Active Raid" enabled it was not showing Egg/Raid boss in label.

- Added new sprites for raids.
Thanks to @sirlindqvist

- Note:- Disabling Gyms/Raids will make disappear all gyms and then show them after 1-3 sec. This (Bug?) is still there.

* Adding unknown raid images to static/images/raid folder

* Bug fixed
- Where disabling gym while leaving raid on removed all markers and showed them after - secs
- If Raid boss has appeared but map did not scanned yet, it show wrong marker/image

* Legionaries are coming....

Bug Fix: Where it will show non scanned gyms with raids in the middle of night.

* Now toggle is called "Show Active Raids only" and bug fix where it will not show Raids only while disabling Gyms

* enhancement: added colour code for level 5 raid "star"

* Bug fix: mentioned by tomballgithub where unknown raid bosses gym will not show raid level and time remaining

And new unknown static images fro raids.

* linting

* Extract zip on npm run build.

* chmod -x map.js

* Now the Pokemon on sidebar gym info are sorted by deployment_time instead of CP_decay and thier original CP is displayed with color coding that shows how many battle needs to kick a pokemon out of gym.

* updated the values to match with 26.67% loss per battle

* lint error fixed

* changes according to review

* updated to show motivation CP and absolute CP in same container , with CSS.

* Added css classes for the gym name color and motivation color

* added nowrap as asked.

* added missing nowrap for motivation.

* Indentation.

pogo-excalibur committed:
Fix the levelup script. (#2428)

* Fixed startup_db call, and added a hash key check.

* Typo.

pogo-excalibur committed:
Added a default raid icon for Wailmer. (#2432)

SwenWacker committed:
Update shared-config.ini.example to remove SQLite (#2435)

Since #2288 support for SQLite is removed. Update another doc refering to SQLite.

FrostTheFox committed:
Add warning on -novc (#2430)

* add warning on -novc

* Don't display the message if we're an -os

* pep8 fix

* combine ifs

PhoenixHugs committed:
Update speed-scheduler.md (#2431)

* Update speed-scheduler.md

* Update speed-scheduler.md

Tweaked verbiage of comment

* Update speed-scheduler.md

jaake77 committed:
Create CODE_OF_CONDUCT.md (#2429)

* Create CODE_OF_CONDUCT.md

* Update CODE_OF_CONDUCT.md

animanmaster committed:
Sync the dropdown with the stored value on load. (#2439)

friscoMad committed:
Add device model to remote config request (#2437)

* Add device model to remote config request

* Fix field to send

FrostTheFox committed:
Remove iPhone 5/5c from device info (#2434)

* Remove iPhone 5/5c

* whoops forgot

Kneckter committed:
Update to fix CONSTRAINT_4 error for spawnpoint table (#2416)

* Update models.py

RM tries to write DB records with a 'latest_seen=3600'. Changing the Check statement allows for these records to be written instead of throwing "InternalError(4025, u'CONSTRAINT CONSTRAINT_4 failed for rocketmapdbadmin.spawnpoint')".

* Implement changes and safety measures for <= 3600.

* Travis.

* Added DB migration

* Fixed long lines in SQL commands

sLoPPydrive committed:
Add team_id to raid webhook data (#2359)

...so that notifications can also show the current gym owner team.

Alderon86 committed:
Update webhooks.md. (#2443)

IZODI committed:
Add explanatory gif to bookmarklet page. (#2442)

* add gif to bookmarklet.html for better understanding of what to do

BrownSlaughter committed:
Show raid details even if image is missing. (#2417)

* show raid details with missing pokemonWithImages

* change to single block with variable

* Fix raidImage scope.

friscoMad committed:
Add more error reporting to level-up script. (#2436)

friscoMad committed:
Fix camel case and flake errors in levelup (#2444)

roberthbelter committed:
Add account name in bad scan log (#2446)

* Add account name in bad scan log

* Consistency w/ rest of code.

sebastienvercammen committed:
Add calendar time to deployment time. (#2449)

* Add HH:mm time to deployment time.

* Change format to MMM Do HH:mm.

* Fix gym details CSS.

* Force calendar time on new line.

* Fix typo + use divs like old code.

* Fix old typo in SCSS file.

daangroot committed:
Correct default scan delay in comments. (#2450)

sebastienvercammen committed:
Add option to disable kph and hlvl-kph. (#2452)

* Add option to disable kph and hlvl-kph.

* Added logging.

Also reworked some small details.

* Fix scope and wait.

* Don't wait if kph is disabled.

friscoMad committed:
Send encounter id as string. (#2433)

ClarkyKent committed:
Dynamic loading of Encounter's list and webhook (#2404)

* adding dynamic loading of iv/CP whitelist

* using enc_whitelist_file instead of args and reading file if it has been modified

* Added webhook whitelist and blacklist as well in dynamic loading

* Simplify refreshing.

daangroot committed:
Add toggle to show/hide Pokémon stats. (#2454)

Add toggle to show/hide Pokémon stats.

BrownSlaughter committed:
Fix undefined/null display issue for raid re-scan gap. (#2453)

* fix for issue raid timing gap

Alderon86 committed:
API 0.89.1 (#2456)

pogo-excalibur committed:
Added a default raid icon for Aggron. (#2460)

sebastienvercammen committed:
Account may not be selected yet in exception. (#2462)

tomballgithub committed:
Add total summary at top of status web page across all map instances (#2372)

* Adds total summaries at top of status web page across all map instances

* Avoid division by zero + consistent HTML placing.

darkelement1987 committed:
Update external.md (#2465)

Change Tdp to Tcp

BrownSlaughter committed:
Dynamic rarity (#2459)

* Reduce performance hit by caching result.

* Only check refresh if loaded.

* Simplified query + binary search + rename -rt.

* Allow 0 to use all data.

* Update docs.

* add to faq

* Export to JSON + move rarity to front-end + add config item for rarity update frequency.

* Fix cyclic import.

* Disable by default. Preferred on one -os.

* Allow on -os only.

* Speed up: better querying & indexing.

* Update docs + add opt group + don't require -os.

* Allow dynamic rarity to run on non-ns instances.

sebastienvercammen committed:
devkat: DB upsert performance improvements. (#2464)

* Test base class to query conversion.

Fix variable name.

* Support peewee defaults.

Fix reference to meta defaults.

Fix field defaults.

Fix row reference.

Fix double escaping.

* Use class for fields instead of row.

Reference field values.

Use proper references for peewee.

Make sure all columns are added.

Fix KeyError.

Fix defaults keys.

Fix order of values.

Sort rows.

* Support peewee column name changes.

* Translate input fields.

Removed unnecessary escaping.

Added extra comments.

* Use InsertQuery to model the query.

Pythonic first item of iterator.

Convert callable default to static.

* Only call callables when needed.

* Double step size.

tomballgithub committed:
Add option to Notify by Pokemon Level (#2399)

* Add user interface option to Notify By Pokemon Level

Fixed travis

Change code to handled undefined notification parameters

* Fix typo

* Fix rebase issue

* Make changes based on review input

* Cleanup

* Remove errant const

* Fix conflict in logic.

tomballgithub committed:
Add Level to Notifications when IV info is present (#2371)

billyjbryant committed:
Add costume to Pokemon and costume/form to GymPokemon. (#2461)

* Introduced Costume to Pokemon & GymPokemon, Introduced Form to GymPokemon

* Switched DB changes to migrate()

* Updated Form to always check, not just for 201.
Fixed call for Costume & Form in Gym lookup

* Removing 'pokemon_data' from string

ftrueck committed:
Update de.json (#2455)

* Update de.json

MARTINI1 committed:
Translated moves, types & rarity to Spanish (#2199)

koh001 committed:
Update de.json (#2473)

All attacks in alphabetical order!

koh001 committed:
Translate rarity text. (#2474)

Fixed bug with rarity text without translation

roberthbelter committed:
Update to Api 0.91.1 (#2484)

Sttug committed:
Update pgoapi. (#2486)

Update pgoapi.

FrostTheFox committed:
SSL Docs (#2481)

* Rewrite SSL doc

* Add certbot

* bug fixes

* Remove redundant warning

pogo-excalibur committed:
Updated default raid boss icons. (#2491)

FrostTheFox committed:
fix ssl docs (#2492)

xxsytriskxx commented:
Hey man, sorry didn't know how to contact you other than here. I've been told you've added PGPool support into stock RM, am I correct? If so, can i fork the build off you? :o

Widdin committed:
Update custom.js: add default Pokémon to "Hide Pokémon". (#2471)

* Expanded 'custom.js' with another example.

A lot of people have been asking on how to exclude some Pokémon by default, so having it in the wiki is a good idea.

* Updated to Sebby's request.

* Tiny mistake fix

Alderon86 committed:
Add fetch_all_news to RPC login sequence. (#2494)

Add fetch_all_news to RPC login sequence.

farstarss committed:
Open Google Maps link in app instead of browser. (#2497)

* Changed Google Maps link

Fixed Google Maps opening in browser instead of app when clicking on any coordinates.

* Fixed Google Maps URL

KartulUdus committed:
Potential EX gym detection (#2440)

* Store if gym is in park for possible raids

BrownSlaughter committed:
Add shiny field to db for gym pokemon (#2501)

* Add shiny field to db

javisabalete committed:
Weather (#2414)

* Add images

* Update comments & variable.

FrostTheFox committed:
0.91.2 (#2505)

Sttug committed:
Make Pokémon weather boosted condition a significant WH field. (#2503)

* Make Pokémon weather boosted condition a significant WH field

friscoMad committed:
Add DB execution context for gym_park. (#2508)

sebastienvercammen committed:
Fix TTH count for one-hour spawns. (#2512)

* Fix TTH count.

* Update comments.

fosJoddie committed:
Replace weather images with emojis. (#2502)

* Remove weather images in favor of weather emojis

* fix bad copy pasta of emojis

* rename CSS class to `weather-boost`

123FLO321 committed:
Add Deoxys and Castform's weather forms. (#2488)

* add castform forms

neskk committed:
Database Cleanup (#2281 rework) (#2383)

* add removal of gym and gym details

* Improved cleanup of old HashKeys.

* Fixes to status page (outdated worker status)

Levonestral committed:
Status Page: Add account busy/idle to instances. (#2479)

* Added idle/busy to instances and total header.

daangroot committed:
Use pokedex submodule for pokemon and move data. (#2510)

* Add pokedex submodule, update gruntfile and update docs.

* Change bluemix git clone instruction.

sebastienvercammen committed:
Update gender on encounter. (#2513)

friscoMad committed:
Only spin stops with levelup script (#2509)

* Only spin stops

* Fix code :D

sebastienvercammen committed:
Change encounter data update. (#2516)

sebastienvercammen committed:
Re-add #2513. (#2517)

sebastienvercammen committed:
Improvements to level-up script. (#2519)

* Improvements to level-up script.

* Suggest a pokéstop if none are in spin range.

Alderon86 committed:
Typo in tutorial.md (#2520)

sebastienvercammen committed:
Exit if no cells are in the geofenced area. (#2521)

pogo-excalibur committed:
Add default icons for new raid bosses. (#2522)

* Updated default raid boss icons.

* 122 -> 121.

roberthbelter committed:
Log correct amount of dropped items. (#2525)

daangroot committed:
Fix typo in -DCf docs. (#2529)

* Fix clean up forts documentation.

* Update utils.py

FrostTheFox committed:
Remove old/unused tools. (#2531)

contadordeestorias committed:
Update fr.json. (#2361)

* Update fr.json

* Sort alphabetically.

wuotan76 committed:
Update ko.json. (#2493)

* Update ko.json.

* Sort file alphabetically.

roberthbelter committed:
Add Log Filename (#2463)

* Add Log Filename

* Allow date and status name formatting.

sebastienvercammen committed:
Update pgoapi: Pin protobuf to 3.5.1. (#2536)

Pin protobuf to 3.5.1.

sebastienvercammen committed:
Improve --dump censor. (#2534)

* Improve --dump censor.

* Censor shared_config.

Alderon86 committed:
Remove Trainer table and old db migration from RocketMap. (#2533)

* Remove Trainer Name and Level from RocketMap

* remove old db migration code

sebastienvercammen committed:
Allow custom hashing service + higher key values. (#2537)

* Allow custom hashing service + higher key values.

* Make service case insensitive.

* Require specific choices.

sebastienvercammen committed:
When hashing quota is exceeded, sleep until quota gets reset. (#2535)

* Sleep until RPM gets reset.

* Limit the sleep to a max.

jonaharagon committed:
Fix Markdown Syntax on DigitalOcean Guide (#2539)

Missing ` on line 28

sebastienvercammen committed:
Support -hs in level-up tool. (#2543)

animanmaster committed:
Efficiently exclude hidden Pokemon (#2540)

* Sync the dropdown with the stored value on load.

* Efficiently filter out excluded ids using a NOT IN query rather than an in-memory filter.

* Fix travis findings and address PR feedback.

* Only process eids if pokemon=true

* Use set comprehension to simplify eid set generation.

sebastienvercammen committed:
Optionally rely on BossLand headers. (#2546)
